crates/bevy_color/src/color.rs

```diff
@@ -1,26 +1,36 @@
-use crate::{Alpha, Hsla, Lcha, LinearRgba, Oklaba, Srgba, StandardColor, Xyza};
+use crate::{Alpha, Hsla, Hsva, Hwba, Laba, Lcha, LinearRgba, Oklaba, Srgba, StandardColor, Xyza};
 use bevy_reflect::{Reflect, ReflectDeserialize, ReflectSerialize};
 use serde::{Deserialize, Serialize};
 
-/// An enumerated type that can represent any of the color types in this crate.
-///
-/// This is useful when you need to store a color in a data structure that can't be generic over
-/// the color type.
-#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize, Reflect)]
-#[reflect(PartialEq, Serialize, Deserialize)]
-pub enum Color {
-    /// A color in the sRGB color space with alpha.
-    Srgba(Srgba),
-    /// A color in the linear sRGB color space with alpha.
-    LinearRgba(LinearRgba),
-    /// A color in the HSL color space with alpha.
-    Hsla(Hsla),
-    /// A color in the LCH color space with alpha.
-    Lcha(Lcha),
-    /// A color in the Oklaba color space with alpha.
-    Oklaba(Oklaba),
-    /// A color in the XYZ color space with alpha.
-    Xyza(Xyza),
+use macros::define_color;
+
+define_color! {
```

Comment:
This macro usage is going to be controversial. I personally think we should revert it, both in the interests of not getting bogged down and in terms of maintainability.

Reply:
That's part of the reason why I put this up as a draft now, just to get feedback on this shortcut. I'm happy to remove both macros